components/system/components/GlobalCarousel.js

```diff
@@ -124,6 +125,12 @@ export class GlobalCarousel extends React.Component {
     window.removeEventListener("slate-global-close-carousel", this._handleClose);
   };
 
+  componentDidUpdate = (prevProps) => {
+    if (this.state.index >= (this.props.objects?.length || 0)) {
```

Comment:
Nice use of the language feature! Still feel like this condition is a little brittle, can we improve it?